users/vieting/experiments/switchboard/ctc/feat/transducer_system_v2.py

```diff
@@ -923,10 +923,19 @@ def run_test_recog_step(
             )
 
     def run_align_step(self, align_args: Optional[Dict] = None) -> None:
-        for train_exp_name in self.returnn_configs.keys():
+        align_args = align_args or {}
+        train_exp_name = align_args.pop('train_exp_name', None)
+        if train_exp_name is None:
+            for train_exp_name in self.returnn_configs.keys():
+                for al_c in self.align_corpora:
+                    self._nn_alignment(
+                        train_exp_name,
+                        align_corpus_key=al_c,
+                        **(align_args or {}),
+        else:  
```

Comment:
Can't we just keep it like that? It shouldn't be too much overhead, right?

Reply:
If I pass the train_exp_name in align_args, I cannot also pass it explicitly. I have to change the run_align_step function in some way to handle this. Not sure whats best yet.

Reply:
In my view, the change in line 864 is sufficient. Then you can later just pick the alignment that you want based on the respective `train_exp_name`.

Reply:
The change leads to an key error. I think I have to remove the outer loop in here